site/docs/v1/tech/lambdas/self-service-registration.adoc

```diff
@@ -0,0 +1,70 @@
+---
+layout: doc
+title: Self Service Registration Validation
+description: An overview of the self-service registration validation lambda
+navcategory: customization
+---
+
+:sectnumlevels: 0
+
+== Self Service Registration Validation lambda
+
+When you have an link:/docs/v1/tech/guides/advanced-registration-forms[Advanced Registration Form] you may add a lambda
+to perform additional validation of the form fields as the user completes steps in the form.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
+
+[source,javascript]
+----
+function validate(result, user, registration, formContext) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `validate` that takes four parameters parameters. The parameters that the lambda is passed are:
+
+* `result` - this is a FormValidationResult object with one property
+  - `errors` which is a FusionAuth Errors object to append form errors to, and is the output of the function
+* `user` - the FusionAuth User object
+* `registration` - the FusionAuth UserRegistration object
+* `formContext` - an object containing data about the current form state. If has the following properties:
+  - `fields` - an array of FormFields in the current form step
+  - `form` - the FusionAuth Form object
+  - `step` - the current step number
+  - `stepIndex` - the zero-indexed number of the current step (always step - 1)
+  - `totalSteps` - the total number of steps in the form
+
+The `result` Errors object is documented in the link:/docs/v1/tech/apis/errors[Errors API]
+
+The `user` object is documented in the link:/docs/v1/tech/apis/users[User API]
+
+The `registration` object is documented in the link:/docs/v1/tech/apis/registrations[Registration API]
+
+The `fields` object is an array of `fields` is an array of link:/docs/v1/tech/apis/form-fields[Form Fields]
+
+The `form` object is documented in the link:/docs/v1/tech/apis/forms[Forms API]
+
+and `form` is documented in the link:/docs/v1/tech/apis/forms[Forms API].
+
+With this information you may programatically perform any validation you wish as the user steps through the self-service
+registration form.
```

Comment:
I'm still not quite sure what I do. Do I look at the form values and add errors?